proposals/dictionary-expressions.md

```diff
@@ -231,22 +293,36 @@ This would allow `ImmutableDictionary<TKey, TValue>` to be annotated with `[Coll
+Pointing at the existing method would have the positive that it would add to the BCL surface area.  But it would have the negative of forcing going through interfaces to create an immutable dictionary.
 
-1. Provide no support (or no support in C# 13), leaving such dictionaries as ones you could not use *dictionary expressions* for.
+Pointing at a new method would allow for more efficient, stack-based, initialization.  However, it would also increase the BCL surface area.
 
-1. Provide a special syntactic form *solely* for the purpose of supplying that value.  For example: `[comparer: myComp, "mads": 21, .. ldmMembers]`.  Here `comparer` would be a contextual keyword.  Users wanting to use that as an actual key would say `@comparer: 1`
+BCL might also prefer something like `CreateRange(ReadOnlySpan<TKey> keys, ReadOnlySpan<TValues> values)` such a signature would then allow constant data segments to be pointed at with an dictionary expression like so:
 
-1. Provide a special syntactic form for the purpose of supplying data to constructors and create methods.  For example: `[new: (comparer: myComp, capacity: 50), "mads": 21, .. ldmMembers]`
+```c#
+FrozenDictionary<int, int> mapping = [1: 2, 2: 4, 4: 8];
+
+// maps to:
+// No stack allocs at all!
+private static ReadOnlySpan<int> __keys => [1, 2, 4];
+private static ReadOnlySpan<int> __values => [2, 4, 8];
+mapping = FrozenDictionary.CreateRange(__keys, __values);
+```
 
 ## Construction
 
 > The elements of a collection expression are evaluated in order, left to right. Each element is evaluated exactly once, and any further references to the elements refer to the results of this initial evaluation.
-> 
+>
+> If the collection expression includes a `comparer: expression`, that `expression` will be evaluated before the subsequence elements are evaluated, and will be converted to an `IEqualityComparer<TKey>`.  If present:
+>
+> 1. If using a constructor to instantiate the value, the constructor must take an `IEqualityComparer<TKey>` as the sole parameter which the converted expression will be passed to.
+> 2. If using a `create method`, the method must take an `IEqualityComparer<TKey>` as the first parameter which the converted parameter will be passed to.
+> 3. If creating an interface, the converted expression will be used as the equality comparer controlling the behavior of the final type (synthesized or otherwise) instantiated.
+>
 > ```diff
-> + A dictionary_element evaluates its interior expressions in order,
+> + A key_value_pair_element evaluates its interior expressions in order,
 > + left to right.  In other words, the key is evaluated before the value. 
 > ```
```

Comment:
@stephentoub  for thoughts.  Specifically, would you prefer we don't change the surface area of the BCL and the compiler just redirects to the existing CreateRange methods that take an `IEnumerable<KVP>`?  Or is there appetite for more efficient overloads that you would add that woudl allow efficient stack-allocing of keys and values (or KVPs), and passing those along? 

We defer to you here.